src/lib.mo

```diff
@@ -21,20 +21,13 @@ module {
 
   public type PostFunc<S, R> = S -> ?R;
 
-  func debugMessage(name : ?Text, key : ?Text) : Text {
-    let name_part = switch (name) {
-      case (?t) "Method name: " # t # ".";
-      case null "";
-    };
-    let key_part = switch (key) {
-      case (?t) "Key of response: " # t # ".";
-      case null "";
-    };
-    if (name_part != "" and key_part != "") name_part # " " # key_part else name_part # key_part;
+  func debugMessage(name : ?Text, index : Nat) : Text {
+    let ?n = name else return "";
+    "Method name: " # n # ". Index: " # Nat.toText(index) # ".";
   };
 
-  class Response<T, S, R>(lock_ : Bool, pre_ : PreFunc<T, S>, post_ : PostFunc<S, R>, name : ?Text, key : ?Text, limit : Nat) {
-    let debug_message : Text = debugMessage(name, key);
+  class Response<T, S, R>(lock_ : Bool, pre_ : PreFunc<T, S>, post_ : PostFunc<S, R>, debug_ : Bool, name : ?Text, index : Nat, limit : Nat) {

```

Comment:
I would combine name and index into one argument `name : Text` and pass that name in as a whole (i.e. call `debugMessage` outside). Response doesn't have to know about index.

We can even combine the `debug_` flag into it as well and make it one argument `debugMsg : ?Text`. Then from outside we pass null instead of `debug_ = false`. The only thing preventing us from doing that is the last line:
```
if (Option.isNull(result)) throw Error.reject("Response rejected. " # debug_message);
```
If we want a single argument `debugMsg : ?Text` then in the null case we can't have a per-response text in Error.reject. Which may also be fine. Maybe we need the pre-response text in Error.reject only if we also need the verbose debug output.